### ISSUE TITLE
Regarding `--` behavior

### ISSUE THREAD

**Issue by verdverm** [gilramir/argparse#1]
From: https://github.com/gilramir/argparse#notes

> If the parser sees "--" on the command-line, it denotes the beginning of a positional argument, and no other switch arguments will be processed.

Normally this stops all parsing and passes any remaining args on, for example, I have an argparse subcommand that invokes an external command, and needs some of the args & flags to come through, as we just pass them through.

What do you think about having argparse stop parsing completely if it reaches a `--`?

**Comment by gilramir** [comment]
You're the first person I've heard from that uses argparse, besides me. I'm happy to hear you're either using it or looking to use it. Now I need to be more careful about breaking things and version numbering!

Can you clarify what you mean by "passes any remaining args on"?  I mean, your program needs to know the passed through arguments somehow, so it can then use them when constructing the arguments for the external command.

Are you talking about putting arguments after "--" into a separate variable?  Can you show a little of what you want to do and how you might expect it to work?

**Comment by verdverm** [comment]
Hmm, it seems I may not be clear on what the prevailing usage of `--` is...

My understanding was that it stops arg/flag parsing for the command, but it seems it is just flag parsing.

I'm currently using spf13/pflags & cobra for a CLI, found argparse because I was looking for something lighter weight for something within it. I have a situation where I need to specify args & flags to two commands together, so I can say "create/modify this thing here, this other thing does the work"


```
dash new playground tail --scope p0.a1:final -- eval test.cue *.json -l data:
```

Maybe with both your argparse and cobra/pflag, the command structure consumes a number of non-flag inputs until finds a leaf or runnable command, then passes any remaining args into that callback.

To be honest, I think my question is probably moot at this point. You should have this operate the standard way, which it seems it probably is, and I learned something by opening this PR.

**Comment by gilramir** [comment]
Right, I started argparser for 2 reasons... one; it fit my conceptual model of arguments more, since I spent so many years using argparse in Python. And second, yes, it's mutch lighter weight than cobra.

Yes, "--" is used to denote the end of optional arguments, so that an argument that starts with a dash won't be confused for an argument. Let's take a look at a python example:

```
import argparse

parser = argparse.ArgumentParser()

parser.add_argument("-e")
parser.add_argument("rest", nargs="*")

args = parser.parse_args()
print("-e:   ", args.e)
print("rest: ", args.rest)
```

and then running it:
```
gram@penguin:~$ python3 test.py 
-e:    None
rest:  []
gram@penguin:~$ python3 test.py  -e 1
-e:    1
rest:  []
gram@penguin:~$ python3 test.py  -e 1 -e 2
-e:    2
rest:  []
gram@penguin:~$ python3 test.py  -e 1 -- -e 2
-e:    1
rest:  ['-e', '2']
```

See also: https://www.cyberciti.biz/faq/what-does-double-dash-mean-in-ssh-command/

The string tokens still need to be gathered by your application (dash) so that it can run the things you asked it to run.

If you have any more questions or requests, especially on things that I didn't complete 100% (like support for i18n), keep on asking!
